### PR TITLE
TRITON-687 node-triton-metrics should support listening on a socket

### PR DESCRIPTION
**Pull request by dyep49**

### CODE REVIEW

File: lib/metrics-manager.js [hunk 1]
```diff
@@ -109,6 +114,7 @@ function MetricsManager(config) {
     this.gcCounts = {};
     this.gcPauses = {};
     this.previousCpuUsage = {user: 0, system: 0};
+    this.path = config.path;
```
Comment (joshwilsdon) [inline lib/metrics-manager.js]: alphabetically I'd expect this.path before this.previousCpuUsage. :)

File: lib/metrics-manager.js [hunk 2]
```diff
@@ -94,6 +92,13 @@ function MetricsManager(config) {
     assert.string(config.staticLabels.server, 'config.staticLabels.server');
     assert.string(config.staticLabels.service, 'config.staticLabels.service');
 
+    if (config.port) {
+        assert.number(config.port, 'config.port');
+        assert.string(config.address, 'config.address');
+    } else {
+        assert.string(config.path, 'config.path');
```
Comment (joshwilsdon) [inline lib/metrics-manager.js]: should we assert that address is *not* set when path is set? We're passing self.address into self.server.listen later, are we expecting that to be undefined?

File: test/metrics-manager.test.js [hunk 3]
```diff
@@ -223,6 +224,80 @@ test('create arbitrary metrics group', function(t) {
...
+    socketPath: serverPath
+});
+
+var socketMetricsManager;
+
+test('setup socket server', function(t) {
+    // Cleanup old server if last test run failed before teardown
+    fs.unlink(serverPath, function unlinked(err) {
+        if (err && err.code !== 'ENOENT') {
+            t.error(err);
```
Comment (joshwilsdon) [inline test/metrics-manager.test.js:240]: should this also t.end and return? Or is continuing safe here?

**Review by joshwilsdon — COMMENTED** [review]
I think this looks pretty good overall. I added a few questions/nits for you to take a look at. Nothing critical though.

File: lib/metrics-manager.js (hunk 1 above)
Comment (dyep49) [reply lib/metrics-manager.js]: Fixed

File: lib/metrics-manager.js (hunk 2 above)
Comment (dyep49) [reply lib/metrics-manager.js]: Fixed

File: test/metrics-manager.test.js (hunk 3 above)
Comment (dyep49) [reply test/metrics-manager.test.js:240]: Fixed

**Review by joshwilsdon — APPROVED** [review]
looks good to me!

**Pull request merged**